From a241414157c5a8b1badf4580350fa9cf45ed498a Mon Sep 17 00:00:00 2001
From: Srinu Jella <sjella@codeaurora.org>
Date: Thu, 20 Aug 2015 14:57:31 +0530
Subject: [PATCH 1/8] Bluetooth: Handle Vendor Specific cmd complete event
 properly

As the Host look for proper opcode in the commad complete case,
and incase of Vendor specific commands SOC doesn't respond with
proper opcode, this has been taken care to avoid time out.

CRs-Fixed: 846052
Change-Id: I65e5b5a9dfeb7a641239c4931faeb2f1da3a20da
Signed-off-by: Kshitij Gupta <kshitijgm@gmail.com>
Signed-off-by: ittat <ittat@live.com>
---
 hci/src/hci_layer.cc | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/hci/src/hci_layer.cc b/hci/src/hci_layer.cc
index ed2908cdd..61ca7de68 100644
--- a/hci/src/hci_layer.cc
+++ b/hci/src/hci_layer.cc
@@ -666,8 +666,16 @@ static waiting_command_t* get_waiting_command(command_opcode_t opcode) {
     waiting_command_t* wait_entry =
         reinterpret_cast<waiting_command_t*>(list_node(node));
 
-    if (!wait_entry || wait_entry->opcode != opcode) continue;
-
+    if (!wait_entry || wait_entry->opcode != opcode) {
+        if(((wait_entry->opcode & HCI_GRP_VENDOR_SPECIFIC) == HCI_GRP_VENDOR_SPECIFIC) &&
+           ((opcode & HCI_GRP_VENDOR_SPECIFIC) == HCI_GRP_VENDOR_SPECIFIC)) {
+            LOG_DEBUG("%s VS event found treat it as valid 0x%x", __func__, opcode);
+        }
+        else {
+            continue;
+        }
+    }
+ 
     list_remove(commands_pending_response, wait_entry);
 
     return wait_entry;
-- 
2.17.1

